flink-ml-benchmark/README.md

```diff
@@ -0,0 +1,172 @@
+Program execution finished
+Job with JobID a5d8868d808eecfb357eb904c961c3bf has finished.
+Job Runtime: 897 ms
+Accumulator Results: 
+- numElements (java.lang.Long): 10000
+
+
+{
+  "name" : "KMeansModel-1",
+  "totalTimeMs" : 897.0,
+  "inputRecordNum" : 10000,
+  "inputThroughput" : 11148.272017837235,
+  "outputRecordNum" : 10000,
+  "outputThroughput" : 11148.272017837235
+}
+```
+
+The command above would save the results into `./output/results.json` as below.
+
+```json
+[ {
+  "name" : "KMeansModel-1",
+  "totalTimeMs" : 897.0,
+  "inputRecordNum" : 10000,
+  "inputThroughput" : 11148.272017837235,
+  "outputRecordNum" : 10000,
+  "outputThroughput" : 11148.272017837235
+} ]
+```
+
+## Customize Benchmark Configuration
+
+`flink-ml-benchmark.sh` parses benchmarks to be executed according to the input
+configuration file, like `./examples/benchmark-example-conf.json`. It can also
+parse your custom configuration file so long as it contains a JSON object in the
+following format.
+
+- The file should contain the following as the metadata of the JSON object.
+  - `"version"`: The version of the json format. Currently its value must be 1.
+- Keys in the JSON object, except `"version"`, are regarded as the names of the
+  benchmarks.
+- The value of each benchmark name should be a JSON object containing the
+  following keys.
+  - `"stage"`: The stage to be benchmarked.
+  - `"inputs"`: The input data of the stage to be benchmarked.
+  - `"modelData"`(Optional): The model data of the stage to be benchmarked, if
+    the stage is a `Model` and needs to have its model data explicitly set.
+- The value of `"stage"`, `"inputs"` or `"modelData"` should be a JSON object
+  containing the following keys.
+  - `"className"`: The full classpath of a `WithParams` subclass. For `"stage"`,
+    the class should be a subclass of `Stage`. For `"inputs"` or `"modelData"`,
+    the class should be a subclass of `DataGenerator`.
+  - `"paramMap"`: A JSON object containing the parameters related to the
+    specific `Stage` or `DataGenerator`.
+
+Combining the format requirements above, the example configuration in
+`./examples/benchmark-example-conf.json` is as follows.
+
+```json
+{
+  "version": 1,
+  "KMeansModel-1": {
+    "stage": {
+      "className": "org.apache.flink.ml.clustering.kmeans.KMeansModel",
+      "paramMap": {
+        "featuresCol": "features",
+        "k": 2,
+        "distanceMeasure": "euclidean",
+        "predictionCol": "prediction"
+      }
+    },
+    "modelData":  {
+      "className": "org.apache.flink.ml.benchmark.clustering.kmeans.KMeansModelDataGenerator",
+      "paramMap": {
+        "seed": null,
+        "numValues": 1,
+        "arraySize": 2,
+        "vectorDim": 10
+      }
+    },
+    "inputs": {
+      "className": "org.apache.flink.ml.benchmark.data.DenseVectorGenerator",
+      "paramMap": {
+        "seed": null,
+        "outputCols": ["features"],
```

Comment:
There is no input for data generator. So the `output` in the name seems redundant.

How about naming it as `columnNames`?